java/piranha/src/test/resources/com/uber/piranha/XPMethodChainCases.java

```diff
@@ -0,0 +1,94 @@
+package com.uber.piranha;
+
+class XPMethodChainCases {
+
+  interface SomeParam {
+    @BoolParam(key = "other_flag")
+    BoolParameter otherFlag();
+
+    // Annotation and the abstract method should be removed
+    @BoolParam(key = "stale_flag")
+    BoolParameter staleFlag();
+
+    static SomeParam create(Parameter cp) {
+      return null;
+    }
+  }
+
+  public static void foobar(Parameter cp) {
+    SomeParam sp = SomeParam.create(cp);
+    // Matches API
+    if (sp.staleFlag().getValue()) {
```

Comment:
nice. let us add one more condition on `otherFlag`. 

```
if (sp.otherFlag().getValue()) { .... }
```

Another case is to add a boolean expression 

```
if (sp.otherFlag().getValue() && sp.staleFlag().getValue()) { .... } 
```
Just to make sure that partial evaluation continues to hold.

Reply:
✅